src/runtime/hexagon/device/hexagon_cdsprpc_shim.cc

```diff
@@ -0,0 +1,178 @@
+/*!
+ * Copyright 2019 Qualcomm Innovation Center, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE‐2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * This Contribution is being provided by Qualcomm Technologies, Inc.,
+ * a Delaware corporation, or its subsidiary Qualcomm Innovation Center, Inc.,
+ * a California corporation, under certain additional terms and conditions
+ * pursuant to Section 5 of the Apache 2.0 license.  In this regard, with
+ * respect to this Contribution, the term "Work" in Section 1 of the
+ * Apache 2.0 license means only the specific subdirectory within the TVM repo
+ * (currently at https://github.com/dmlc/tvm) to which this Contribution is
+ * made.
+ * In any case, this submission is "Not a Contribution" with respect to its
+ * permitted use with any of the "vta" and "verilog" subdirectories in the TVM
+ * repo.
+ */
+#ifdef __ANDROID__
+#include <android/log.h>
+#include <dlfcn.h>
+#include <stdint.h>
+#include <sys/stat.h>
+
+#include "android_Release_aarch64/remote.h"
+
+#define WEAK __attribute__((weak))
+
+namespace {
+
+void* LoadLibCdspRpc() {
```

Comment:
consider make The CdspRPC a class(CDSPRPCSession, put all the fields into the class, and use Global singleton pattern). The redirection can get global singleton and run.